src/test/java/io/jenkins/plugins/analysis/warnings/IssuesRecorderITest.java

```diff
@@ -71,21 +88,388 @@ public void shouldCreateUnstableResult() {
         assertThat(result).hasTotalSize(8);
         assertThat(result).hasOverallResult(Result.UNSTABLE);
 
-        HtmlPage page = getWebPage(result);
+        HtmlPage page = getWebPage(result, "eclipseResult");
         assertThat(page.getElementsByIdAndOrName("statistics")).hasSize(1);
     }
 
-    private HtmlPage getWebPage(final AnalysisResult result) {
+    /**
+     * returns a WebPage-instance of an analysis result.
+     * @param result the corresponding AnalysisResult of a finished build.
+     * @param page the relative path to the result page.
+     * @return a WebPage as {@link HtmlPage}.
+     */
+    private HtmlPage getWebPage(final AnalysisResult result, final String page) {
         try {
             WebClient webClient = j.createWebClient();
-            webClient.setJavaScriptEnabled(false);
-            return webClient.getPage(result.getOwner(), "eclipseResult");
+            //webClient.setJavaScriptEnabled(false);
+            return webClient.getPage(result.getOwner(), page);
         }
         catch (SAXException | IOException e) {
-           throw new AssertionError(e);
+            throw new AssertionError(e);
+        }
+    }
+
+    /**
+     * Verifies that the right amount of duplicate code warnings are detected.
+     */
+    @Test
+    public void shouldHaveDuplicateCodeWarnings(){
+        FreeStyleProject project = createJobWithWorkspaceFile("duplicateCode/cpd.xml");
+        Cpd cpd = new Cpd();
+        enableWarnings(project, cpd);
+
+        AnalysisResult result = scheduleBuildAndAssertStatus(project, Result.SUCCESS);
+
+        assertThat(result).hasTotalSize(20);
+        assertThat(result).hasOverallResult(Result.SUCCESS);
+
+    }
+
+    /**
+     * Verifies that the priority of the duplicate code warnings are changed corresponding
+     * to the defined thresholds for cpd warnings.
+     */
+    @Test
+    public void priorityShouldChangeIfThresholdsChange() {
+        FreeStyleProject project = createJobWithWorkspaceFile("duplicateCode/cpd.xml");
+        Cpd cpd = new Cpd();
+        cpd.setNormalThreshold(1);
+        cpd.setHighThreshold(2);
+        enableWarnings(project, cpd);
+
+        List<HtmlTableRow> tableRows = getIssueTableRows(project);
+        HtmlTableRow row = tableRows.get(0);
+        List<HtmlTableCell> tableCells = row.getCells();
+
+        assertThat(tableCells.get(2).getTextContent()).isEqualTo("High");
+
+
+        cpd.setHighThreshold(5);
+
+        tableRows = getIssueTableRows(project);
+        row = tableRows.get(0);
+        tableCells = row.getCells();
+
+        assertThat(tableCells.get(2).getTextContent()).isEqualTo("Normal");
+
+
+        cpd.setNormalThreshold(4);
+
+        tableRows = getIssueTableRows(project);
+        row = tableRows.get(0);
+        tableCells = row.getCells();
+
+        assertThat(tableCells.get(2).getTextContent()).isEqualTo("Low");
+    }
```

Comment:
Die 5 Zeilen (156-162, usw.) jeweils in eine Methode auslagern mit Parameter: threshold, expectedPriority.